ui/src/components/TabbedLayout.tsx

```diff
@@ -1,14 +1,46 @@
-import React from 'react';
 import {PageSection, PageSectionVariants, Tab, Tabs, TabTitleText,} from '@patternfly/react-core';
 import {DepCompoundTable} from "./DepCompoundTable";
 import {getSourceName, getSources, Report} from "../api/report";
+import {AnalyticsBrowser, AnalyticsBrowserSettings} from '@segment/analytics-next'
+import React, {useEffect, useRef} from 'react';
+import {useAppContext} from '../App';
 
 export const TabbedLayout = ({report}: { report: Report }) => {
+  const appContext = useAppContext();
   const sources = getSources(report);
-
   const [activeTabKey, setActiveTabKey] = React.useState<string | number>(getSourceName(sources[0]));
   const [isTabsLightScheme] = React.useState<boolean>(true);
 
+  const analytics = appContext.writeKey && appContext.writeKey.trim() !== '' ?
+      AnalyticsBrowser.load({ writeKey: appContext.writeKey } as AnalyticsBrowserSettings) : null;
+  const previousUserId = useRef<string | null>(null);
+  const previousActiveTabKey = useRef<string | number>('');
+
+    useEffect(() => {
+      if (!analytics) return;
+
+      if (appContext.userId == null) {
+        if (appContext.anonymousId != null) {
+          analytics.setAnonymousId(appContext.anonymousId);
+        }
+      } else {
+        if (appContext.userId !== previousUserId.current) {
+          analytics.identify(appContext.userId);
+          previousUserId.current = appContext.userId;
+        }
+      }
+      const handleActiveTabKeyUpdate = async (newActiveTabKey: string | number) => {
+        if (newActiveTabKey !== previousActiveTabKey.current) {
+          analytics.track("rhda.exhort.tab", {
```

Comment:
Can't you set the userId/anonymousId to the track event as we do in the backend?
Try using a `SegmentEvent` instead:
```
analytics.track({
  userId: appContext.userId,
  anonymousId: appContext.anonymousId,
  event: "rhda.exhort.tab"
}, {
  tabName: newActiveTabKey
});
```

I haven't tried it, it's just from what I saw in the JS api.

Reply:
It does not change the value of it if I add it, only adds it as part of the property. Identify() changes the userId and setAnonymousId() changes anonymousId.